lib/shopify/response.ex

```diff
@@ -7,15 +7,16 @@ defmodule Shopify.Response do
 
   defstruct [
     :code,
-    :data
+    :data,
+    :headers
   ]
 
-  def new(code, body, resource) when code < 300 do
-    {:ok, %Response{code: code, data: resource |> parse_json(body)}}
+  def new(%{body: body, code: code, headers: headers}, resource) when code < 300 do
```

Comment:
My only concern with this is that its a breaking API change. Technically we are still sub 1.0, so this isnt a huge deal, but ideally, we keep it as is for now, and potentially update as we enter 1.0 (eventually?).

So - thoughts on:

```elixir
def new(code, body, resource_or_error, headers \\ [])
```

Having the headers at the end is a little strange, but it maintains the API for now.

Reply:
Sure, I'll update this. Today will be busy for me so I probably won't get to it until tomorrow.

Heads up that `Shopify.Response` is marked `@moduledoc false` which [indicates that this module is not part of the public API of the library](https://hexdocs.pm/elixir/writing-documentation.html#hiding-internal-modules-and-functions).

Reply:
You're right - I didn't notice that here. Thanks for the heads up. Ignore my request then.